x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/structurefinder/GrokPatternCreator.java

```diff
@@ -822,8 +822,11 @@ public boolean matchesAll(Collection<String> sampleMessages, TimeoutChecker time
             TimeoutChecker timeoutChecker
         ) {
 
-            explanation.add("A full message Grok pattern [" + grokPattern.substring(2, grokPattern.length() - 1) + "] looks appropriate");
-
+            if (grokPattern.startsWith("%{") && grokPattern.endsWith("}")) {
```

Comment:
is it ever the case that this pattern contains regex with it as well as grok captures?

Reply:
Only if it's supplied by the end user, and in that case this explanation doesn't make sense.  This explanation was originally added on the assumption it would relate to one of the "full line" matches on lines 54-68 of this file.  Later on the same class was used to apply arbitrary stuff the user might have supplied, but in that case they know that's the only Grok pattern that will be considered.